src/mumble/widgets/EventFilters.cpp

```diff
@@ -152,3 +152,24 @@ bool SkipFocusEventFilter::eventFilter(QObject *obj, QEvent *event) {
 
 	return QObject::eventFilter(obj, event);
 }
+
+FocusEventObserver::FocusEventObserver(QObject *parent, bool consume) : QObject(parent), m_consume(consume) {
+}
+
+bool FocusEventObserver::eventFilter(QObject *obj, QEvent *event) {
+	QWidget *widget = static_cast< QWidget * >(obj);
+
+	if (widget && event->type() == QEvent::FocusAboutToChange) {
+		QFocusEvent *focusEvent = static_cast< QFocusEvent * >(event);
+
+		if (!widget->hasFocus() && focusEvent->gotFocus()) {
+			emit focusInObserved(focusEvent->reason());
+		} else if (widget->hasFocus() && !focusEvent->gotFocus()) {
+			emit focusOutObserved(focusEvent->reason());
+		}
+
+		return m_consume;
```

Comment:
I assume we wouldn't want to consume the event in case neither of these two if-clauses are executed (which at least in theory is possible, given the way the code is written)? :thinking:

Reply:
Ah yes! Good catch! And yes, focus events seem to be very glitchy in Qt...